File: DACN/backend_src/app/models/shift.py
from app.database import Base
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Time, Boolean
from sqlalchemy.orm import relationship

class Shift(Base):
    __tablename__ = "shifts"
    id = Column(Integer, primary_key=True, index=True)
    employee_id = Column(Integer, ForeignKey("employees.id", ondelete="CASCADE"), nullable=False)
    date = Column(DateTime, nullable=False)
    start_time = Column(Time, nullable=False)
    end_time = Column(Time, nullable=False)
    is_overtime = Column(Boolean, default=False, nullable=False)
    overtime_note = Column(String(500), nullable=True)
    # employee = relationship("Employee")  # Commented to avoid circular import

File: DACN/Models/Shift.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using System.ComponentModel.DataAnnotations.Schema;
namespace Models
{
    [Table("shifts")]
    public class Shift
    {
        [Key]
        public int Id { get; set; }
    [Column("employee_id")]
    public int EmployeeId { get; set; }
        public DateTime? Date { get; set; }
    [Column("start_time")]
    public TimeSpan? StartTime { get; set; }
    [Column("end_time")]
    public TimeSpan? EndTime { get; set; }
    [Column("is_overtime")]
    public bool IsOvertime { get; set; } = false;
    [Column("overtime_note")]
    [StringLength(500)]
    public string? OvertimeNote { get; set; }
    }
}
